lib/workers/repository/update/pr/body/config-description.ts

```diff
@@ -23,9 +26,16 @@ export function getPrConfigDescription(config: BranchConfig): string {
   }
   prBody += '\n\n';
   prBody += emojify(':recycle: **Rebasing**: ');
-  if (config.rebaseWhen === 'behind-base-branch') {
+  const rebaseWhen =
+    config.rebaseWhen === 'auto'
+      ? config.automerge === true ||
+        (await platform.getBranchForceRebase?.(config.baseBranch))
+        ? 'behind-base-branch'
+        : 'conflicted'
+      : config.rebaseWhen;
```

Comment:
Is this intended to to reflect the same result as from https://github.com/renovatebot/renovate/blob/2d2d0b9e735625a46568234d8924dff40784a273/lib/workers/repository/update/branch/reuse.ts#L46-L52 ?

Reply:
Yes, I duplicated the logic from there. We could factor it out into a helper function if you prefer.

Reply:
Is it possible to _store_ that result and pass it through? Instead of calling it twice.

Reply:
I think we could memoize the resolved value of the returned promise of `platform.getBranchForceRebase(...)`, so the first time this function is called and the promise is resolved, the value is cached and subsequent calls will take the cached value instead of recomputing it. To avoid implementing this memoization for each platform implementation, I could try adding it in [`platform.initPlatform(...)`](https://github.com/renovatebot/renovate/blob/7392dbe63179ae6aaad0b9dc1b582a1beb277b06/lib/modules/platform/index.ts#L43). WDYT?

Reply:
I meant that I prefer to save the result into the config which is passed through

Reply:
Would you mind sketching a bit more concretely what you envision the implementation to look like? `platform.getBranchForceRebase(...)` has only the base branch name as argument, so this function cannot update the config without changing its signature. Also, each platform implements this function, so a naive implementation would duplicate the config updating logic, which violates the DRY principle. Alternatively, we could introduce a helper function which calls `platform.getBranchForceRebase(...)` and updates the config, but we'd need to make sure that always the helper function and not `platform.getBrancheForceRebase(...)` is called –this feels a bit error-prone IMO. That's why I thought to add a transparent memoization wrapper in `platform.initPlatform(...)`.

Reply:
Current code is this:

```
  if (
    config.rebaseWhen === 'behind-base-branch' ||
    (await shouldKeepUpdated(config, baseBranch, branchName)) ||
    (config.rebaseWhen === 'auto' &&
      (config.automerge === true ||
        (await platform.getBranchForceRebase?.(config.baseBranch))))
  ) {
```

Could it instead be changed to something like this?

```
  if (config.rebaseWhen === 'auto') {
    if (config.automerge === true) {
      logger.debug('Converting rebaseWhen=auto to rebaseWhen=behind-base-branch because automerge=true');
      config.rebaseWhen = 'behind-base-branch';
    }
    if (await platform.getBranchForceRebase?.(config.baseBranch)) {
      logger.debug('Converting rebaseWhen=auto to rebaseWhen=behind-base-branch because platform is configured to require up-to-date branches');
      config.rebaseWhen = 'behind-base-branch';
    }
  }
  if (
    config.rebaseWhen === 'behind-base-branch' ||
    (await shouldKeepUpdated(config, baseBranch, branchName))
  ) {
```

Reply:
I see. Yes, I believe that could work. _But_ it makes a precise rebase description highly dependent on the execution order, i.e. it will only be precise when `shouldReuseExistingBranch(...)` is called before `getPrConfigDescription(...)`, which doesn't feel too robust. If you prefer this implementation over some kind of memoization of `platform.getBranchForceRebase(...)`, I can of course do it.

Reply:
This is part of branch logic, which always comes before PR logic, so we should make sure it's deterministic

Reply:
Okay, I've updated `shouldReuseExistingBranch(...)` along the lines of what you suggested: 856d36b WDYT?

Reply:
Have you tested it on your example repo(s) to see if it satisfies your requirements still?

Reply:
No, and as it turned out: good point! I've pushed another commit that fixes retaining the updated config. Now, I've tested it on a real repo and the rebase description is properly derived.